sofdev6-02-2024/LosAderezos
# Scan Bar code with camera and with image

Testing steps:
1. Remove container from docker desktop
2. docker compose up --build
3. Login to the application at http://localhost:5173/
4. Go to the Products page http://localhost:5173/products/
5. Click on the button to scan Barcode

![image](https://github.com/user-attachments/assets/58300488-396f-4495-998b-04af582f8e99)

6. A modal will appear where you have two options

Scan code with camera

1. First of all in Google Chrome go to the top left of the link and give permissions to camera.
![Screenshot from 2024-11-24 14-00-47](https://github.com/user-attachments/assets/d487dfef-f929-4bc7-b55d-cc2f68b4b822)
2. Click on the Start Scan button that will enable the camera in the center of the modal.
3. Print a medium size barcode to improve the scanning, I didn't test if it was recognized on a cell phone, and center it on the camera.
4, After a while if it is well centered and it scans it will be redirected to the Single Product Page of the specific product. 

Scan code with image
1. Click on the Attach Image button.
2. The content will update and you will be able to choose a .jpeg or .png image.
3. Click the CHoose file button and enter an image from the generated ones. 
4. Click on the Scan image button.


Use the inspect to see everything that is happening and if there is any error it will be shown there, the scan image functionality has a problem that I still do not know what it can be if you can help me with that better.
